lib/embiggen/uri.rb

```diff
@@ -37,46 +29,22 @@ def shortened?
 
     private
 
-    def check_redirects(redirects)
-      return unless redirects.zero?
-
-      fail TooManyRedirects.new("#{uri} redirected too many times", uri)
-    end
-
-    def check_location(location)
-      return if location
+    def extract_redirects(request_options = {})
+      redirects = request_options.fetch(:redirects) { Configuration.redirects }
+      fail TooManyRedirects.new(
+        "#{uri} redirected too many times", uri) if redirects.zero?
```

Comment:
That's true: it's the last URI followed.

Perhaps it's just the message that needs correcting here as the initial URI is in the original `Embiggen::URI` instance but it seems useful to show how much progress was made.